source/plugins/ruby/CAdvisorMetricsAPIClient.rb

```diff
@@ -876,9 +881,14 @@ def getNodeLastRebootTimeMetric(metricJSON, hostName, metricNametoReturn, metric
 
         metricCollection = {}
         metricCollection["CounterName"] = metricNametoReturn
-        #Read it from /proc/uptime
-        metricCollection["Value"] = DateTime.parse(metricTime).to_time.to_i - IO.read("/proc/uptime").split[0].to_f
-
+        if !@os_type.nil? && !@os_type.empty? && @os_type.strip.casecmp("windows") == 0 && ExtensionUtils.isAADMSIAuthMode()
+          #Read from the windowsnodereset.log for Windows daemonset with MSI auth mode using Windows AMA
+          metricValStr = IO.readlines("C:\\etc\\kubernetes\\host\\windowsnodereset.log")[0].split[0]
```

Comment:
Questions 1) what happens either file doesnt exist or cant be read? 2) is this file exist on both Ws2019 & ws 2022? 3)  how large this file can be? 4)  do we need to trim for whitespaces or newlines etc?

Suggestions 1) define file name as constant 2) add the example of the content 3) wrap them begin & rescue to handle any corner case 4)  handle the fail to read the file scenario.